orchagent/aclorch.cpp

```diff
@@ -765,13 +760,80 @@ shared_ptr<AclRule> AclRule::makeShared(acl_table_type_t type, AclOrch *acl, Mir
     throw runtime_error("Wrong combination of table type and action in rule " + rule);
 }
 
+bool AclRule::enableCounter()
+{
+    SWSS_LOG_ENTER();
+
+    if (m_ruleOid == SAI_NULL_OBJECT_ID)
+    {
+        SWSS_LOG_ERROR("ACL rule %s doesn't exist in ACL table %s", m_id.c_str(), m_tableId.c_str());
+        return false;
+    }
+
+    if (!createCounter())
+    {
+        return false;
+    }
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_ACL_ENTRY_ATTR_ACTION_COUNTER;
+    attr.value.aclaction.parameter.oid = m_counterOid;
+    attr.value.aclaction.enable = true;
+
+    sai_status_t status = sai_acl_api->set_acl_entry_attribute(m_ruleOid, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to enable counter for ACL rule %s in ACL table %s", m_id.c_str(), m_tableId.c_str());
+        return false;
+    }
+
+    return true;
+}
+
+bool AclRule::disableCounter()
+{
+    SWSS_LOG_ENTER();
+
+    if (m_ruleOid == SAI_NULL_OBJECT_ID)
+    {
+        SWSS_LOG_ERROR("ACL rule %s doesn't exist in ACL table %s", m_id.c_str(), m_tableId.c_str());
+        return false;
+    }
+
+    sai_attribute_t attr;
+
```

Comment:
Do we need to check the existance of counter before disable it?

Reply:
@bingwang-ms it is not mandatory, however we can save some time on SAI API call

Reply:
@bingwang-ms would be better to remove to save system resources

Reply:
@bingwang-ms it is not mandatory, however we can save some time on SAI API call

Reply:
@bingwang-ms done